tests/approaches/test_grammar_search_invention_approach.py

```diff
@@ -479,7 +491,7 @@ def _evaluate_atom_trajectory(
             return heuristic_fn(atoms_sequence[0])
 
     score_function = _MockHAddLookahead(initial_predicates, atom_dataset,
-        train_tasks, candidates, lookahead_depth=1)
+        train_tasks, candidates, "hadd", lookahead_depth=1)
     assert score_function.evaluate(set(candidates)) == float("inf")
```

Comment:
Let's add tests for the two new approaches? We can just copy the first few tests from `def test_hadd_lookahead_score_function():` (before the depth stuff)